docs/sql/statements/select.rst

```diff
@@ -493,6 +502,15 @@ rows, and is an alternative to the :ref:`LIMIT <sql-select-limit>` clause::
    If ``number_of_results`` is null, then no limit is applied, essentially the
    query is returning all rows, as if not ``FETCH`` clause is present.
 
+.. NOTE::
+
+   ``number_of_results`` can be a :ref:`CAST or TRY CAST <sql-type-cast>`
+   expression. e.g.::
+
+     FETCH FIRST CAST('50' AS long) ROWS ONLY
+     FETCH FIRST TRY_CAST(? AS long) ROWS ONLY
+     FETCH FIRST '100'::int ROWS ONLY
+
 .. NOTE::
```

Comment:
Do you think it would be worth to restructure this a bit to define the `number_of_results` only once and reduce the amount of `NOTE` directives a bit? Not sure if this is good for the reading flow.

Reply:
imo, and since this is a reference section, I believe that is easier to have the notes where they belong instead of referencing them. If we choose to reference them though, I think it would be better to have them in a separate file which is not included in any doc index, and it's only used as an appendix to which we can refer to from any place in the docs. WDYT?

Reply:
It would probably fit into https://crate.io/docs/crate/reference/en/5.2/sql/general/value-expressions.html

This also relates a bit to the parser grammar changes. I think we shouldn't have too many different kind of expressions to keep these explanations simpler. E.g. we could define a "integerExpression" or "numberExpression" once in that document and then refer to it.

Reply:
thx!